jni_gen/lib/src/tools/android_sdk_tools.dart

```diff
@@ -63,10 +63,13 @@ task listDependencies(type: Copy) {
   ///
   /// This function temporarily overwrites the build.gradle file by a stub with
   /// function to list all dependency paths for release variant.
-  /// This function fails if no gradle build is attempted
-  static List<String> getGradleClasspaths() {
-    stderr.writeln('trying to obtain gradle classpaths');
-    const android = 'android';
+  /// This function fails if no gradle build is attempted before.
+  ///
+  /// if current project is not directly buildable by gradle, eg: a plugin,
+  /// a relative path to other project can be specified using [androidProject].
+  static List<String> getGradleClasspaths([String androidProject = '.']) {
+    stderr.writeln('trying to obtain gradle classpaths...');
+    final android = join(androidProject, 'android');
```

Comment:
```dart
      throw Exception('\n\ngradle exited with exit code ${procRes.exitCode}\n'
          'This can be related to a known issue with gradle. '
          'Please run `flutter build apk` and try again\n');
```

->

```dart
      throw Exception('\n\ngradle exited with exit code ${procRes.exitCode}\n'
          'This can be related to a known issue with gradle. '
          'Please run `flutter build apk` in $androidProject and try again\n');
```